narwhal/primary/src/synchronizer.rs

```diff
@@ -654,12 +715,13 @@ impl Synchronizer {
         // Instruct workers to download any missing batches referenced in this certificate.
         // Since this header got certified, we are sure that all the data it refers to (ie. its batches and its parents) are available.
         // We can thus continue the processing of the certificate without blocking on batch synchronization.
-        let inner = self.inner.clone();
         let header = certificate.header().clone();
         let max_age = self.inner.gc_depth.saturating_sub(1);
-        self.inner.batch_tasks.lock().spawn(async move {
-            Synchronizer::sync_batches_internal(inner, &header, max_age, true).await
-        });
+        self.inner
+            .tx_batch_tasks
+            .send((header.clone(), max_age, true))
```

Comment:
Since `certified` is always true, is it necessary to send it over?

Reply:
Ah I've kept it this way in order to have the `tx_batch_tasks` for more "generic" use, but maybe not needed. I'll refactor